application/frontend/controllers/MfaController.php

```diff
@@ -40,7 +42,13 @@ public function actionCreate()
 
         $label = $req->getBodyParam('label');
 
-        return Mfa::create($user->id, $type, $label);
+        // rpOrigin is needed for WebAuthn authentication
+        $rpOrigin = $req->get('rporigin', '');
+        if ($rpOrigin != '' && !in_array($rpOrigin, \Yii::$app->params['authorizedRPOrigins'])){
```

Comment:
I'm wondering if this reused compound `if` statement/block should be a separate mini-function, or maybe
make the params key a constant?

Reply:
yeah maybe

Reply:
This might not make a difference, by PHP's `in_array()` has a third parameter for if you want it to do strict comparisons, which are generally recommended.
```suggestion
        if ($rpOrigin != '' && !in_array($rpOrigin, \Yii::$app->params['authorizedRPOrigins'], true)){
```